src/btree/bt_delete.c

```diff
@@ -164,6 +164,18 @@ __wt_delete_page(WT_SESSION_IMPL *session, WT_REF *ref, bool *skipp)
         goto err;
     if (addr.ta.prepare)
         goto err;
+
+    /*
+     * When performing a truncate operation with no associated timestamp, limit fast-truncate to
+     * pages where all its data is globally visible. Technically we don't need to check the newest
+     * stop durable timestamp, but for consistency, we check for the maximum of both the start and
```

Comment:
Could we add to the comment that we do this check to prevent data in the history store that should be cleared to reappear?

Reply:
Yes, I can add that in.

Reply:
Fixed in [b96f150](https://github.com/wiredtiger/wiredtiger/pull/10456/commits/b96f150655e2dd3e4319bd9901b313e2c870db39)